.github/workflows/build-test.yml

```diff
@@ -25,19 +31,30 @@ jobs:
 
       - name: Build
         id: build
-        run: mvn -B -U -Dsurefire.rerunFailingTestsCount=5 clean install
+        run: mvn -B -U "-Dsurefire.rerunFailingTestsCount=5" clean install
 
       - name: Archive Test Results on Failure
         uses: actions/upload-artifact@v2
-        if: failure()
+        if: always()
         with:
-          name: test-results
+          name: Unit Test Results ${{ matrix.os }}
           path: target/surefire-reports/
           retention-days: 7
```

Comment:
❓ Previously this artifact was only uploaded on failures, why does it have to happen as well for success? Is it so that the test results can be published or so? It's probably unclear to me because they use different paths.